File: hackerrank/30 days of code/day 25, running time and complexity/day_25_running_time_and_complexity.java
import java.util.*;

public class Solution {
    public static void main(String[] args) {
        var sc = new Scanner(System.in);
        
        for (var n = sc.nextInt(); n-- > 0; ) {
            var number = sc.nextInt();

            if (number == 1) {
                System.out.println("Not prime");
                continue;
            }

            int i = 2, limit = (int) Math.sqrt(number);
            for (; i <= limit; i++)
                if (number % i == 0) {
                    System.out.println("Not prime");
                    break;
                }

            if (i > limit)
                System.out.println("Prime");
        }
    }
}

File: hackerrank/30 days of code/day 25, running time and complexity/day_25_running_time_and_complexity.rb
gets.chomp.to_i.times do
    number = gets.chomp.to_i

    if number == 1
        puts "Not prime"
        next
    end

    i, limit = 2, Math.sqrt(number)
    while i <= limit
        if number.modulo(i).zero?
            puts "Not prime"
            break
        end
        i += 1
    end

    if i > limit
        puts "Prime"
    end
end
